core/src/main/java/com/microsoft/applicationinsights/internal/shutdown/SDKShutdownActivity.java

```diff
@@ -65,6 +68,15 @@ public synchronized void register(Closeable closeable) {
             closeables.add(closeable);
         }
 
+        public synchronized void register(final ExecutorService service) {
```

Comment:
Why registration needs to be synchronized? I think at the end what what matters is that all the stoppables are present in the collection. I don't think there is dependency on specific operation or the order of registration. It might be wise to instead make the list atomic in this case if it is after all needed.

Reply:
Mainly, I was following the pattern of the other `register` methods. Should be fine

Reply:
Sounds good. #Resolved for consistency.